irohad/ametsuchi/reconnection/impl/k_times_reconnection_strategy.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ametsuchi/reconnection/impl/k_times_reconnection_strategy.hpp"
+
+using namespace iroha::ametsuchi;
+
+KTimesReconnectionStorageStrategy::KTimesReconnectionStorageStrategy(
+    size_t number_of_recalls)
+    : number_of_recalls_(number_of_recalls) {}
+
+bool KTimesReconnectionStorageStrategy::canInvoke(const Tag &tag) {
+  CollectionType::accessor accessor;
+  while (true) {
+    bool is_found = invokes_.find(accessor, tag);
+    if (is_found) {
+      // assume the tag is unique and doesn't share between threads
+      accessor->second++;
+    } else {
+      bool is_inserted = invokes_.insert(accessor, tag);
+      if (is_inserted) {
+        accessor->second = 1;
+      } else {
+        continue;
```

Comment:
I find the loop a little unclear. don't you think it could be like something this?
```c++
  bool is_found = false;
  bool is_inserted = false;
  while (not is_found and not is_inserted) {
    if (is_found = invokes_.find(accessor, tag)) {
      // assume the tag is unique and doesn't share between threads
      accessor->second++;
    } else if (is_inserted = invokes_.insert(accessor, tag)) {
      accessor->second = 1;
    }
  }
```

Reply:
Looks great. Thanks.